cmd/stripe/main.go

```diff
@@ -2,10 +2,27 @@ package main
 
 import (
 	"context"
+	"net/http"
+	"sync"
+	"time"
 
 	"github.com/stripe/stripe-cli/pkg/cmd"
+	"github.com/stripe/stripe-cli/pkg/stripe"
 )
 
 func main() {
-	cmd.Execute(context.Background())
+	ctx := context.Background()
+
+	// Set up the telemetry client and add it to the context
+	waitGroup := &sync.WaitGroup{}
+	httpClient := &http.Client{
+		Timeout: time.Second * 3,
+	}
+	telemetryClient := &stripe.AnalyticsTelemetryClient{WG: waitGroup, HTTPClient: httpClient}
+	contextWithTelemetry := context.WithValue(ctx, stripe.TelemetryClientKey{}, telemetryClient)
+
+	cmd.Execute(contextWithTelemetry)
+
+	// Wait for all telemetry calls to finish before existing the process
+	waitGroup.Wait()
```

Comment:
Wait groups can be tricky to use right, but I think you've actually got it right! However, I think it could be simplified and also not leak so much of its implementation.

If you updated the type of `AnalyticsTelemetryClient.WG` to be `sync.WaitGroup` instead of `*sync.WaitGroup`, then you could skip initializing it above, because the zero value of the wait group is usable by default. Then this line could either become `telemetryClient.WG.Wait()`, or you could even add a custom method to telemetry client (Wait? Drain?) and then keep the waitgroup internal.

Reply:
That makes sense to me, thanks!